### PR TITLE
Migrate param_group testing to OptimizerInfo

### PR DESCRIPTION
**Pull request by janeyx99**
Today, our param_group testing does the equivalent of pitting weight and bias with different optimizer hyperparams and then check that the overall result is going the right direction based on maximize.

This PR introduces two tests to encompass coverage:
1. For every optimizer input (no differentiable), always force bias to have 0 weight_decay, and then check that the direction is expected. This is basically a replica to today's tests, but is more methodical as the test is a real use case. 
2. To ensure that the different groups have distinct behavior, I added another test where lr is basically 0 in default group, and ensure that the param in the default group doesn't move while loss does.

Together, these tests do a better job of testing param groups than today's tests, **though we do lose some flavors**. For example, RMSProp also pits centered=True vs False across the param_groups, Adadelta has a variation on rho, and ASGD has a variation for t0. I don't think this is really a loss, as the previous test was just testing for direction and our new tests test stronger guarantees.

The leftover param group configs are used in conjunction with LRSchedulers.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117675

### CODE REVIEW

File: test/test_optim.py [hunk 1]
```diff
@@ -326,6 +326,90 @@ def test_fused_large_tensor(self, device, dtype, optim_info):
...
+
+            weight = Parameter(torch.randn((10, 5), device=device, dtype=dtype))
+            bias = Parameter(torch.randn((10), device=device, dtype=dtype))
+            input = torch.randn(5, device=device, dtype=dtype)
+
+            optimizer = optim_cls([dict(params=[weight], **weight_kwargs), dict(params=[bias], **bias_kwargs)])
+
+            loss = (weight.mv(input) + bias).pow(2).sum()
+            initial_value = loss.item()
+            for _ in range(200):
```
Comment (albanD) [inline test/test_optim.py:347]: How slow is this test, is 200 really necessary here or would 20 work out fine?

File: test/test_optim.py [hunk 2]
```diff
@@ -326,6 +326,90 @@ def test_fused_large_tensor(self, device, dtype, optim_info):
...
+    @optims(optim_db, dtypes=[torch.float32])
+    def test_param_groups_lr(self, device, dtype, optim_info):
+        optim_cls = optim_info.optim_cls
+        # Skip differentiable testing for now, see https://github.com/pytorch/pytorch/issues/116490
+        all_optim_inputs = _get_optim_inputs_including_global_cliquey_kwargs(device, dtype, optim_info, skip=("differentiable",))
+        for optim_input in all_optim_inputs:
+            # optim_input.kwargs will be the param group kwargs, which should have >0 lr
+            if "lr" not in optim_input.kwargs or optim_input.kwargs["lr"] == 0:
+                optim_input.kwargs["lr"] = 1e-3
+            outer_kwargs = {"lr": 1e-28}
```
Comment (albanD) [inline test/test_optim.py:374]: Why not 0 ?

File: test/test_optim.py [hunk 3]
```diff
@@ -326,6 +326,90 @@ def test_fused_large_tensor(self, device, dtype, optim_info):
...
+            irrelevant = Parameter(torch.randn(2, device=device, dtype=dtype))
+            irrelevant_clone = irrelevant.clone()
+            input = torch.randn(5, device=device, dtype=dtype)
+            optimizer = optim_cls(
+                [dict(params=[weight, bias], **optim_input.kwargs), dict(params=[irrelevant])],
+                **outer_kwargs)
+
+            loss = (weight.mv(input) + bias).pow(2).sum()
+            initial_value = loss.item()
+            for _ in range(200):
```
Comment (albanD) [inline test/test_optim.py:390]: Less?

**Review by albanD — APPROVED** [review]
Small questions but SGTM

File: test/test_optim.py (hunk 2 above)
Comment (janeyx99) [reply test/test_optim.py:374]: Some optimizers will divide by a number derived from the lr and we don't want divide by 0 issues.

File: test/test_optim.py (hunk 1 above)
Comment (janeyx99) [reply test/test_optim.py:347]: 20 would work out fine

File: test/test_optim.py [hunk 4]
```diff
@@ -326,6 +326,90 @@ def test_fused_large_tensor(self, device, dtype, optim_info):
...
+
+            weight = Parameter(torch.randn((10, 5), device=device, dtype=dtype))
+            bias = Parameter(torch.randn((10), device=device, dtype=dtype))
+            input = torch.randn(5, device=device, dtype=dtype)
+
+            optimizer = optim_cls([dict(params=[weight], **weight_kwargs), dict(params=[bias], **bias_kwargs)])
+
+            loss = (weight.mv(input) + bias).pow(2).sum()
+            initial_value = loss.item()
+            for _ in range(20):
```
Comment (janeyx99) [inline test/test_optim.py:347]: with 200, the tests take ~112s. moving to 20 => 30s.  for the record.
